### PR TITLE
Using reserve for optimize inserts and minor fixes

### PR DESCRIPTION
**Pull request by GermanAizek**
## Summary
Useful changes, additional insert were found at `emplace_back`, but memory didnt allocated for last insert after loop.

### CODE REVIEW

File: plugins/arclite/archive.hpp [hunk 1]
```diff
@@ -169,7 +169,7 @@ class ArcAPI {
   static const ArcCodecs& codecs() {
     return get()->arc_codecs;
   }
-  static size_t Count7zCodecs() {
+  static const size_t Count7zCodecs() {
```
Comment (alabuzhev) [inline plugins/arclite/archive.hpp:172]: const is useless on return types.

File: plugins/arclite/plugin.cpp [hunk 2]
```diff
@@ -702,8 +702,9 @@ class Plugin
       });
       const auto error_log = std::make_shared<ErrorLog>();
       size_t im = 0, nm = matched_indices.size();
+	  std::vector<UInt32> indices;
```
Comment (alabuzhev) [inline plugins/arclite/plugin.cpp:705]: This changes the semantics: previously on each iteration the vector was recreated, now it's outside the loop and will accumulate all the values.
If you want to optimize here, add `indices.clear()` to the loop.

File: plugins/arclite/archive.hpp (hunk 1 above)
Comment (GermanAizek) [reply plugins/arclite/archive.hpp]: fix it here https://github.com/FarGroup/FarManager/pull/723/commits/d832463aa807c429ca15719bcc7cb5939f306fbd

File: plugins/arclite/plugin.cpp (hunk 2 above)
Comment (GermanAizek) [reply plugins/arclite/plugin.cpp]: u ar absolutely right, I have changed original behavior function.
fix it here https://github.com/FarGroup/FarManager/commit/d832463aa807c429ca15719bcc7cb5939f306fbd

**Comment by GermanAizek** [comment]
@alabuzhev,
I fixed code tabulation by making it like in your other project sources.

File: plugins/arclite/archive.cpp [hunk 3]
```diff
@@ -233,6 +233,7 @@ class MyCompressCodecsInfo : public ICompressCodecsInfo, public IHashers, privat
       }
       if (arc_lib.ComHashers) {
         UInt32 numHashers = arc_lib.ComHashers->GetNumHashers();
+        hashers_.reserve(numHashers);
```
Comment (alabuzhev) [inline plugins/arclite/archive.cpp:236]: Thinking about it, all this code is a part of a bigger loop, and calling `reserve` in a loop is a known antipattern: it is too easy to pessimise exponential growth down to linear.

I suggest leaving it as is, it's not like like this place is a bottleneck anyway.

File: plugins/arclite/archive.cpp [hunk 4]
```diff
@@ -400,6 +401,7 @@ void ArcAPI::load_codecs(const std::wstring& path) {
   const auto& add_hashers = [this](ArcLib &arc_lib, size_t lib_index) {
     if (arc_lib.ComHashers) {
       UInt32 numHashers = arc_lib.ComHashers->GetNumHashers();
+      arc_hashers.reserve(numHashers);
```
Comment (alabuzhev) [inline plugins/arclite/archive.cpp:404]: Same as above: `add_hashers` is calling from a `for` loop below, and then again from a `while` loop.
`reserve`ing here would effectively kill exponential growth,

File: plugins/arclite/plugin.cpp [hunk 5]
```diff
@@ -702,8 +702,11 @@ class Plugin
       });
       const auto error_log = std::make_shared<ErrorLog>();
       size_t im = 0, nm = matched_indices.size();
+      std::vector<UInt32> indices;
+      indices.reserve(nm);
       while (im < nm) {
-        std::vector<UInt32> indices;
+        if (!indices.empty())
```
Comment (alabuzhev) [inline plugins/arclite/plugin.cpp:708]: This condition is redundant, `clear` checks it internally.

**Pull request merged**